### PR TITLE
sync policy doc to reality

### PR DESCRIPTION
**Pull request by deads2k**
The actual implementation has drifted from policy.md.  This update helps reflect reality.

### CODE REVIEW

**Comment by deads2k** [comment]
@smarterclayton @derekwaynecarr  I'd like to make sure that the description for the AccessReviews meet expectations before I update my upstream pull.

File: docs/proposals/policy.md [hunk 1]
```diff
@@ -83,38 +84,37 @@ That covers most cases where cluster admins want to create projects and delegate
...
-	// Verbs is a list of Verbs that apply to ALL the ResourceKinds and AttributeRestrictions contained in this rule.  "*" represents all kinds.
-	Verbs []Verb
+	// Verbs is a list of Verbs that apply to ALL the ResourceKinds and AttributeRestrictions contained in this rule.  VerbAll represents all kinds.
+	Verbs []string
 	// AttributeRestrictions will vary depending on what the Authorizer/AuthorizationAttributeBuilder pair supports.
 	// If the Authorizer does not recognize how to handle the AttributeRestrictions, the Authorizer should report an error.
-	AttributeRestrictions runtime.EmbeddedObject
-	// ResourceKinds is a list of kinds this rule applies to.  "*" represents all kinds.
-	ResourceKinds []string
+	AttributeRestrictions kruntime.EmbeddedObject
```
Comment (smarterclayton) [inline docs/proposals/policy.md]: runtime

File: docs/proposals/policy.md [hunk 2]
```diff
@@ -83,38 +84,37 @@ That covers most cases where cluster admins want to create projects and delegate
...
+	// Resources is a list of resources this rule applies to.  ResourceAll represents all resources.
+	Resources []string
+	// ResourceNames is an optional white list of names that the rule applies to.  An empty set means that everything is allowed.
+	ResourceNames kutil.StringSet
```
Comment (smarterclayton) [inline docs/proposals/policy.md]: util

File: docs/proposals/policy.md [hunk 3]
```diff
@@ -122,56 +122,53 @@ type Role struct {
 	kapi.TypeMeta
 	kapi.ObjectMeta
 
+	// Rules holds all the PolicyRules for this Role
 	Rules []PolicyRule
 }
 
-// RoleBinding references a Role, but not contain it.  It adds who and namespace information.  
-// It can reference any Role in the same namespace or in the master namespace.
+// RoleBinding references a Role, but not contain it.  It adds who and namespace information.
```
Comment (smarterclayton) [inline docs/proposals/policy.md]: This comment is bad - fix in the source

Comment (deads2k) [inline docs/proposals/policy.md]: Clarified

**Comment by deads2k** [comment]
Updated for comments and updated more fields that I hope to change this week.

**Comment by smarterclayton** [comment]
LGTM, squash and you can merge

**Comment by deads2k** [comment]
[merge]

**Pull request merged**